common-custom-user-data-maven-extension/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -47,12 +48,19 @@ private static void captureIde(BuildScanApi buildScan) {
             } else if (sysProperty("eclipse.buildId").isPresent()) {
                 buildScan.tag("Eclipse");
                 buildScan.value("Eclipse version", sysProperty("eclipse.buildId").get());
+            } else if (firstSysPropertyKeyStartingWith("idea.version").isPresent()) {
+                buildScan.tag("IntelliJ IDEA");
+                buildScan.value("IntelliJ IDEA version", stripPrefix("idea.version", firstSysPropertyKeyStartingWith("idea.version").get()));
```

Comment:
I don't understand how this logic works for newer versions of IDEA. there is no code here that reads the actual value of the sys prop idea.version. Or am I missing something?

Reply:
I now realize that there are two IDEA blocks. I think they can be merged into one as it was before.

Reply:
I think the problem here is that in case idea.version sys prop is present, both IDEA blocks will be executed.

Reply:
No, it is not, since there is an if/else statement.